service/Abstractions/Pipeline/DataPipeline.cs

```diff
@@ -268,6 +275,13 @@ public string GetHandlerOutputFileName(IPipelineStepHandler handler, int index =
     [JsonPropertyName("files")]
     public List<FileDetails> Files { get; set; } = new();
 
+    /// <summary>
+    /// The Erorr logs that caused the pipeline to fail, if any.
+    /// </summary>
+    [JsonPropertyOrder(19)]
+    [JsonPropertyName("logs")]
+    public string? Logs { get; set; } = null;
```

Comment:
ditto about consistency. I think a better approach would be assigning an approximate completion time and using that to establish if the pipeline failed.